packages/curvenote-cli/src/directives/listing.ts

```diff
@@ -0,0 +1,71 @@
+import type { DirectiveData } from 'myst-common';
+import type { VFile } from 'vfile';
+
+export const listingDirective = {
+  name: 'listing',
+  doc: 'A listing directive that can be used to show a list of articles from a specific venue, collection or kind.',
+  arg: {
+    type: 'myst',
+    doc: 'The title of the listing.',
+  },
+  body: {
+    type: 'myst',
+    doc: 'Descriptive content to be displayed along with the listing.',
+  },
+  options: {
+    venue: {
+      type: 'string',
+      doc: 'The venue to list articles from.',
+      required: true,
+    },
+    collection: {
+      type: 'string',
+      doc: 'The collection to list articles from.',
+      required: false,
+    },
+    status: {
+      type: 'string',
+      doc: 'The status of articles to list (published | in-review).',
+      required: false,
+    },
+    limit: {
+      type: 'number',
+      doc: 'The maximum number of articles to list.',
+      required: false,
+    },
+    kind: {
```

Comment:
We have usually reserved kind for a sub-type (admonition > warning, proof > example, container > figure). This could be `display: cards` or `class: cards`